hazelcast-wm/src/main/java/com/hazelcast/web/ClusteredSessionService.java

```diff
@@ -104,18 +103,14 @@ public ClusteredSessionService(FilterConfig filterConfig, Properties properties,
 
         es.scheduleWithFixedDelay(new Runnable() {
             public void run() {
-                while (ensureInstanceRunning) {
-                    try {
-                        ensureInstance();
-                    } catch (Exception ignored) {
-                        if (ignored instanceof InterruptedException) {
-                            ensureInstanceRunning = false;
-                        }
-                        EmptyStatement.ignore(ignored);
-                    }
+                try {
+                    ensureInstance();
+                } catch (Exception ignored) {
+                    EmptyStatement.ignore(ignored);
                 }
             }
         }, 0, CLUSTER_CHECK_INTERVAL, TimeUnit.SECONDS);
```

Comment:
Are you aware that after `es.scheduleWithFixedDelay()` is called the method `ensureInstance()` is called again? Because `es.scheduleWithFixedDelay()` is called with an initial delay of `0` the method `ensureInstance()` is effectively called twice when the service is created: once synchronously in the constructor and once asynchronously through the executor. 

Maybe it is better to change the ordering: First do it synchronously and then schedule the thread with an initial delay of `CLUSTER_CHECK_INTERVAL` instead of `0`